pkg/providers/aws/cluster_network_provider.go

```diff
@@ -55,10 +61,55 @@ func (n *NetworkProvider) CreateNetwork() error {
 
 		- Check if a VPC exists with the integreatly.org/clusterID tag
 		- If VPC doesn't exist, create a VPC with CIDR block and tag it
+
+		verify there is a _network strategy
+		note we need to handle the tier differently to other products
 	*/
 	logger := n.logger.WithField("action", "CreateNetwork")
-	logger.Debug("CreateNetwork stub")
-	return errorUtil.New("CreateNetwork stub")
+	logger.Debug("CreateNetwork called")
+
+	clusterID, err := resources.GetClusterID(ctx, c)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "error getting clusterID")
+	}
+	organizationTag := resources.GetOrganizationTag()
+
+	//check if there is an rhmi specific vpc already created.
+	foundVpc, err := getRHMIVpc(ctx, c, ec2Svc, logger, clusterID, organizationTag)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "unable to get vpc")
+	}
+	if foundVpc != nil {
+		return foundVpc, nil
+	}
+
+	createVpcOutput, err := ec2Svc.CreateVpc(&ec2.CreateVpcInput{
+		CidrBlock: aws.String("10.0.0.0/26"),
+	})
```

Comment:
This will be provided through a function parameter or embedded in the NetworkProvider struct, correct?

Reply:
Yes, it's hardcoded for now. Plan is to parameterise it somehow. 👍